framework/Auth/lib/Horde/Auth/Fallback.php

```diff
@@ -0,0 +1,229 @@
+<?php
+/**
+ * Copyright 2016-2017 Horde LLC (http://www.horde.org/)
+ *
+ * See the enclosed file COPYING for license information (LGPL). If you did
+ * not receive this file, http://www.horde.org/licenses/lgpl21
+ *
+ * @author   Ralf Lang <lang@b1-systems.de>
+ * @category Horde
+ * @license http://www.horde.org/licenses/lgpl21 LGPL-2.1
+ * @package  Auth
+ */
+/**
+ * The Horde_Auth_Fallback class provides a way to combine two separate
+ * drivers for failover and legacy support cases, for example
+ * using ldap, but falling back to a local database
+ * To the user, this driver presents the combined, unique users of both backends
+ * Only the primary driver allows adding, editing and removing users.
+ *
+ * @author   Ralf Lang <lang@b1-systems.de>
+ * @category Horde
+ * @license http://www.horde.org/licenses/lgpl21 LGPL-2.1
+ * @package  Auth
+ */
+class Horde_Auth_Fallback extends Horde_Auth_Base
+{
```

Comment:
If supporting arbitrary numbers of drivers, the driver may be renamed to Horde_Auth_Cascade instead. A native speaker should comment on this though.